DigitalLearningSolutions.Web.Tests/Controllers/TrackingSystem/CourseSetup/AdminFieldsControllerTests.cs

```diff
@@ -51,7 +57,151 @@ public void AdminFields_returns_AdminFields_page_when_appropriate_course_found()
             var result = controller.AdminFields(1);
 
             // Then
-            result.Should().BeViewResult().WithDefaultViewName().ModelAs<DisplayPromptsViewModel>();
+            result.Should().BeViewResult().WithDefaultViewName().ModelAs<AdminFieldsViewModel>();
+        }
+
+        [Test]
+        public void PostEditAdminField_save_calls_correct_methods()
+        {
+            // Given
+            var model = new EditAdminFieldViewModel(1, 1, "Test", "Options", false);
+            const string action = "save";
+
+            A.CallTo(
+                () => customPromptsService.UpdateCustomPromptForCourse(
+                    1,
+                    1,
+                    false,
+                    "Options"
+                )
+            ).DoesNothing();
+
+            // When
+            var result = controller.EditAdminField(model, action);
+
+            // Then
+            A.CallTo(
+                () => customPromptsService.UpdateCustomPromptForCourse(
+                    1,
+                    1,
+                    false,
+                    "Options"
+                )
+            ).MustHaveHappened();
+            result.Should().BeRedirectToActionResult().WithActionName("AdminFields");
+        }
+
+        [Test]
+        public void PostEditAdminField_add_configures_new_answer()
+        {
+            // Given
+            var model = new EditAdminFieldViewModel(1, 1, "Test", "Options", false);
+            const string action = "addPrompt";
+
+            A.CallTo(
+                () => customPromptsService.UpdateCustomPromptForCourse(
+                    1,
+                    1,
+                    false,
+                    "Test"
+                )
+            ).DoesNothing();
+
+            // When
+            var result = controller.EditAdminField(model, action);
+
+            // Then
+            using (new AssertionScope())
+            {
+                result.As<ViewResult>().Model.Should().BeOfType<EditAdminFieldViewModel>();
+                AssertNumberOfConfiguredAnswersOnView(result, 2);
+            }
+        }
+
+        [Test]
+        public void PostEditRegistrationPrompt_delete_removes_configured_answer()
+        {
+            // Given
+            var prompt = new CustomPrompt(1, "Test", "Test\r\nAnswer", false);
+            var model = new EditAdminFieldViewModel(prompt, 1);

```

Comment:
I guess it's fine if the unit tests run, but I would have expected this to use the other constructor

Reply:
Yeah I'm not sure why this constructor works here but not on the other ones, I've changed it to the other constructor for consistency.